lib/NormalModule.js

```diff
@@ -327,6 +330,50 @@ class NormalModule extends Module {
 		};
 		const loaderContext = {
 			version: 2,
+			getOptions: (loaderName, schema) => {
+				const loader = loaderContext.loaders[loaderContext.loaderIndex];
+				let options = {};
+
+				if (loader.options && typeof loader.options === "object") {
+					({ options } = loader);
+				} else if (loader.query) {
+					let { query } = loader;
+
+					if (query.substr(0, 1) !== "?") {
+						throw new WebpackError(
+							"A valid query string should begin with '?'"
+						);
+					}
+
+					query = query.substr(1);
+
+					// Allow to use `?foo=bar` in `options`
+					if (query.substr(0, 1) === "?") {
```

Comment:
Should we support `options: "?foo=bar"`? I can update schema to disable because query contains `??foo=bar`

Reply:
double `??` has a special meaning anyway so `loader??foo=bar` is not legal.

I think we should deprecate support for `options: "string"` at all (in config).

Reply:
No we don't need to support `options: "?foo=bar"`.